official/utils/arg_parsers/parsers.py

```diff
@@ -58,9 +58,36 @@ def __init__(self):
 from __future__ import division
 from __future__ import print_function
 
-
 import argparse
 
+import tensorflow as tf
+
+
+def parse_dtype_info(flags):
+  """Convert dtype string to tf dtype, and set loss_scale default as needed.
+
+  Args:
+    flags: namespace object returned by arg parser.
+
+  Raises:
+    ValueError: If an invalid dtype is provided.
+  """
+  flags.dtype = {
+      "fp16": tf.float16,
+      "float16": tf.float16,
+      "fp32": tf.float32,
+      "float32": tf.float32,
+  }.get(flags.dtype, flags.dtype)
```

Comment:
On second read, I think we should stick with just two options-- fp32, fp16.

Also, let's move this to a module dict, then you can just get DTYPE_MAP.keys() for choices below.